### DISCUSSION TITLE
Does it do Flag-based enums?

### DISCUSSION THREAD

**Discussion by CyberBotX** [joseftw/jos.enumeration#23] [Q&A]
I was looking into possibly using this source generator in a small bit of code I have, and I was wondering if it can do flag-based enums. For instance, this is a standard C# flag-based enum:
```csharp
[Flags]
enum Choice
{
    A = 0x1,
    B = 0x2,
    C = 0x4
}
```
With that, I could have an enum that holds `A | B` and would be the equivalent of 0x3, and if I wanted to check for a flag's existence manually, I could do something like `(enum & C) == C`.

I'm also wondering if the database providers for the source generator would handle flag-based enums as well, assuming the source generator itself does.

Just from a quick glance of the code, it does not appear that the source generator can do flag-based enums, but I was wanting to verify that.

**Comment by joseftw** [comment]
Hello!
No, it does not support flags.

Just curious, does flags work on anything that's NOT an enum?

**Comment by CyberBotX** [comment]
According to https://learn.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-8.0 it is only for enums. But looking at the source code of .NET, it seems that the attribute barely does anything except for affect formatting and string output, it doesn't actually affect things like, for instance, Enum.HasFlag.
